models/Tool/Targeting/TargetGroup.php

```diff
@@ -192,4 +192,24 @@ public function getActive()
     {
         return $this->active;
     }
+
+    /**
+     * @return void
+     */
+    public function delete(): void
+    {
+        $this->dispatchEvent(new TargetGroupEvent($this), TargetGroupEvents::POST_DELETE);
+    }
+
+    /**
+     * @return void
+     */
+    public function save(): void
+    {
+        if ($this->getId()) {
+            $this->dispatchEvent(new TargetGroupEvent($this), TargetGroupEvents::POST_UPDATE);
+        }else{
+            $this->dispatchEvent(new TargetGroupEvent($this), TargetGroupEvents::POST_ADD);
+        }
+    }
```

Comment:
The actual action is missing here. You should call `delete()` and `save()` from the `dao`, I think.